internal/utils/files/files_test.go

```diff
@@ -48,9 +48,30 @@ func TestCollectTestStepFiles(t *testing.T) {
 		},
 	} {
 		t.Run(tt.path, func(t *testing.T) {
-			testStepFiles, err := CollectTestStepFiles(tt.path, testutils.NewTestLogger(t, tt.path))
+			testStepFiles, err := CollectTestStepFiles(tt.path, testutils.NewTestLogger(t, tt.path), nil)
 			require.NoError(t, err)
 			assert.Equal(t, tt.expected, testStepFiles)
 		})
 	}
 }
+
+func TestCollectTestStepFilesWithIgnorePatterns(t *testing.T) {
+	// Test with ignore patterns - should silently skip README.md
+	ignorePatterns := []string{"README*", "*.md"}
+	logger := testutils.NewTestLogger(t, "test_data/with-overrides-with-ignore")
```

Comment:
For this test to be more useful I think we should use a mock logger here and:
- make sure that `Logf("Ignoring.*")` does **not** get called on it if matching pattern is provided
- that it does get called if no matching pattern is provided